### DISCUSSION TITLE
Installation access tokens

### DISCUSSION THREAD

**Discussion by lostinpatterns** [community/community#24453] [Apps, API and Webhooks]
<p>I know you need to generate a JWT token for your Github App in order to get an access token for an installation. However, when you do get an installation access token from</p>
<pre><code>https://api.github.com/app/installations/:installation_id/access_tokens
</code></pre>
<p>, is that token returned an OAuth2 access token or a JWT token?</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/lostinpatterns">@lostinpatterns</a>,</p>
<p>Thanks for being here! That is a great question, you’d need to download the key to your local machine. You’ll use this key to sign a <a href="https://jwt.io/introduction" rel="nofollow noopener">JSON Web Token (JWT)</a> and encode it using the <code>RS256</code>algorithm. </p>

**Comment by rsarosh** [comment]
<p>Keep getting this message</p>
<p>{“message”:"‘Issued at’ claim (‘iat’) must be an Integer representing the time that the assertion was issued",“documentation_url”:“<a href="https://developer.github.com/v3" rel="nofollow noopener">https://developer.github.com/v3</a>”}</p>
<p>I created the time using typescript  new Date().getTime() the numbers are as follows:</p>
<p>{<br>
“iat”: 1553791599468,<br>
“exp”: 932274959680800,<br>
“iss”: xxxx<br>
}</p>

**Comment by rsarosh** [comment, marked as answer]
<p>Realized my mistake, here is the JS script </p>
<p>let iat = Math.round(Date.now() / 1000)<br>
let exp = iat + (9 * 60)<br>
let s = iat + " : " + exp<br>
alert (s)</p>
<p>Take these numbers and paste it in <a href="http://jwt.io" rel="nofollow noopener">jwt.io</a> </p>
